code/frameworks/vue3-vite/src/plugins/vue-component-meta.ts

```diff
@@ -134,8 +134,9 @@ export async function vueComponentMeta(tsconfigPath = 'tsconfig.json'): Promise<
             s.replace('export default ', 'const _sfc_main = ');
             s.append('\nexport default _sfc_main;');
           }
-
-          s.append(`\n;${name}.__docgenInfo = ${JSON.stringify(meta)}`);
+          s.append(`\n;${name}.__docgenInfo = Object.assign(${JSON.stringify(meta)}, {
+            displayName: ${name}.name || ${name}.__name || ${JSON.stringify(meta.displayName)}
+          })`);
```

Comment:
Whilst I agree with the need to read `__name`, I'd prefer if we read it before `meta` and if we honour `meta` if it is defined.

So you could do something like:
```ts
Object.assign({
  displayName: ${name}.name || ${name}.__name,
}, JSON.stringify(meta))
```

This will ensure whatever changes are made in the future to the layer in charge of extracting docgen information will be correctly taken into account here.